zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -205,18 +237,42 @@ impl<F: Field> MPTConfig<F> {
                     ifx! {not!(f!(q_not_first)) => {
                         require!(a!(is_start) => true);
                     }};
+                    cb.base.require_zero(
+                        "only one state is valid at a time",
+                        a!(is_start) * a!(is_branch) * a!(is_account) * a!(is_storage)
+                    );
                     // Main state machine
                     matchx! {
                         a!(is_start) => {
+                            require!(a!(q_node) => 0.expr());
+                            require!((FixedTableTag::StartNode.expr(), a!(q_row)) => @"fixed");
                             state_machine.start_config = StartConfig::configure(meta, &mut cb, ctx.clone());
                         },
                         a!(is_branch) => {
+                            require!((FixedTableTag::BranchNode.expr(), a!(q_row)) => @"fixed");
+                            let diff = a!(q_node)- a!(q_node_prev);
+                            // Start -> Branch || Branch -> Branch
+                            require!(
+                                (diff.clone() - START_NODE_ROWS.expr()) * (diff.clone() - BRANCH_NODE_ROWS.expr()) 
+                                => 0.expr());
```

Comment:
Doing it like this is a bit tricky. I think it's better to just make sure for each node that `a!(is_start), a!(is_branch), a!(is_account) or a!(is_storage)` are set to `1` after `Count` rows. So for example for a branch node, for all following rows up to `ExtensionBranchRowType::Count`, `a!(is_start) + a!(is_branch) + a!(is_account) + a!(is_storage) => 0`, on the `ExtensionBranchRowType::Count`th row however, `a!(is_start) + a!(is_branch) + a!(is_account) + a!(is_storage) => 1`.